enhancements/microshift/microshift-apiserver-certs.md

```diff
@@ -229,24 +229,48 @@ The external network is used for external connections, such as `oc` clients.
 The internal network is used for internal connections, such as the ones from
 pods in the cluster.
 
-Choosing the internal network is a complex task, as it can collide with other
-IP ranges coming from DHCP or other environment configurations. In order to
-minimize risks, the API server service IP, which is always the [first one](https://github.com/kubernetes/kubernetes/blob/master/pkg/controlplane/services.go#L47)
-from the service CIDR range, is selected by default.
+When configuring apiserver we need to provide an IP where it will be listening.
+This turns into an `EndpointSlice` object that will create an association
+between the apiserver service network IP and https port (defaults to
+10.43.0.1:443) and the IP where it will actually be listening on the real port,
+which defaults to 6443.
+Ovnk creates flows to turn this VIP into the one from the `EndpointSlice`. Here
+we need a valid IP, local to the node, where the apiserver is listening so that
+this translation succeeds.
+Imagine we use the same service network apiserver address for the
+`EndpointSlice`: 10.43.0.1. This creates a VIP-endpoint in ovnk that looks like
+10.43.0.1:443->10.43.0.1:6443. Imagine we have an interface in the host where
+the apiserver service network IP has been configured. While regular pod network
+origin traffic goes through the ovnk flows to translate it correctly, this is
+not the case with host network pods. These fall through the flow that redirects
+node IP traffic (which is the source for the host network pods) to the
+apiserver IP, 10.43.0.1, configured in some interface. Since the port is not
+modified by ovnk traffic ends up reaching whoever is listening on port 443 in
+any host interface. And there is a process listening there, the router for
+ingress traffic. Certificate validations fail, and even if they succeeded this
+is not the apiserver but a different service.
```

Comment:
This is clearer.

Eventually we need to update the docs in https://github.com/openshift/microshift/tree/main/docs/network as well. I think some of that information is now in product documentation, too, so we'll want to find that and make sure it is updated, too.

Reply:
There were a [few updates](https://github.com/openshift/microshift/commit/c2129c257c573a49e644be83a99f2bee13219f87) as part of the fix, flows and a small note avout the additional IP. Amyway all this qualifies for product docs now, I will open a ticket for that.